src/main/kotlin/no/nav/familie/ks/sak/kjerne/behandling/steg/vilkårsvurdering/VilkårsvurderingUtils.kt

```diff
@@ -253,3 +262,78 @@ fun finnTilOgMedDato(tilOgMed: LocalDate?, vilkårResultater: List<VilkårResult
 
     return if (skalVidereføresEnMndEkstra) tilOgMed.plusMonths(1).sisteDagIMåned() else tilOgMed.sisteDagIMåned()
 }
+
+fun validerBarnasVilkår(vilkårsvurdering: Vilkårsvurdering, barna: List<Person>) {
+    val feiler = mutableListOf<String>()
+
+    barna.map { barn ->
+        vilkårsvurdering.personResultater
+            .flatMap { it.vilkårResultater }
+            .filter { it.personResultat?.aktør == barn.aktør }
+            .forEach { vilkårResultat ->
+                val fødselsdato = barn.fødselsdato.tilDagMånedÅr()
+                val vilkårType = vilkårResultat.vilkårType
+                if (vilkårResultat.resultat == Resultat.OPPFYLT && vilkårResultat.periodeFom == null) {
+                    feiler.add("Vilkår '$vilkårType' for barn med fødselsdato $fødselsdato mangler fom dato.")
+                }
+                if (vilkårResultat.periodeFom != null && vilkårResultat.lagOgValiderPeriodeFraVilkår().fom.isBefore(barn.fødselsdato)) {
+                    feiler.add(
+                        "Vilkår '$vilkårType' for barn med fødselsdato $fødselsdato " +
+                            "har fra-og-med dato før barnets fødselsdato."
+                    )
+                }
+                if (vilkårResultat.periodeFom != null &&
+                    vilkårResultat.erEksplisittAvslagPåSøknad != true &&
+                    vilkårResultat.vilkårType == Vilkår.MELLOM_1_OG_2_ELLER_ADOPTERT
+                ) {
+                    vilkårResultat.validerVilkår_MELLOM_1_OG_2_ELLER_ADOPTERT(
+                        vilkårResultat.lagOgValiderPeriodeFraVilkår(),
+                        barn.fødselsdato
+                    )?.let { feiler.add(it) }
+                }
+            }
+    }
+
+    if (feiler.isNotEmpty()) {
+        throw Feil(feiler.joinToString(separator = "\n"))
+    }
+}
+
+private fun VilkårResultat.lagOgValiderPeriodeFraVilkår(): IkkeNullbarPeriode<Long> = when {
+    periodeFom !== null -> {
+        IkkeNullbarPeriode(verdi = behandlingId, fom = checkNotNull(periodeFom), tom = periodeTom ?: TIDENES_ENDE)
+    }
+    erEksplisittAvslagPåSøknad == true && periodeTom == null -> {
+        IkkeNullbarPeriode(verdi = behandlingId, fom = TIDENES_MORGEN, tom = TIDENES_ENDE)
+    }
+    else -> {
+        throw FunksjonellFeil("Ugyldig periode. Periode må ha t.o.m.-dato eller være et avslag uten datoer.")
+    }
+}
+
+private fun VilkårResultat.validerVilkår_MELLOM_1_OG_2_ELLER_ADOPTERT(
+    periode: IkkeNullbarPeriode<Long>,
+    barnFødselsdato: LocalDate
+): String? = when {
+    this.erAdopsjon() && periode.tom.isAfter(etterAugustBarnetFyller6År(barnFødselsdato.plusYears(6))) ->
+        "Du kan ikke sette en t.o.m dato som er etter august året barnet fyller 6 år."
+    this.erAdopsjon() && periode.tom.diffIDager(periode.tom) > 365 ->
```

Comment:
```suggestion
    this.erAdopsjon() && periode.fom.diffIDager(periode.tom) > 365 ->
```
akkurat nå sammenligner man to samme dato